script/spec-runner.js

```diff
@@ -230,8 +230,13 @@ async function runMainProcessElectronTests () {
 }
 
 async function installSpecModules (dir) {
+  // v8 headers use c++17 so override the gyp default of -std=c++14,
+  // but don't clobber any other CXXFLAGS that were passed into spec-runner.js
+  const CXXFLAGS = ['-std=c++17', process.env.CXXFLAGS].filter(x => !!x).join(' ');
```

Comment:
IMO this is unnaceptable, if native modules don't work without this setting we **need** a way to provide it to end users without telling everyone to set `CXXFLAGS`, can this be set via `config.gypi` / `common.gypi`.

This is a fundamental breaking change that will cause a lot of issue traffic if it isn't transparent to the user.

Reply:
It looks like the following across various platforms:
* [Linux](https://github.com/nodejs/node-gyp/issues/1543#issuecomment-421452554) `cflags_cc` in `common.gypi`
* [Windows](https://github.com/nodejs/node-gyp/issues/1662#issuecomment-754283275) `LanguageStandard` in `common.gypi`
* [macOS](https://github.com/nodejs/node-gyp/issues/1543#issuecomment-517883775) `OTHER_CFLAGS` in `common.gypi`